src/store/item.js

```diff
@@ -1,4 +1,13 @@
 import { writable } from 'svelte/store';
+import colorOptions from '@/constants/item/colors';
+
+export const colors = writable(colorOptions);
+export const customColors = writable({});
+export function addNewColor(id, color) {
+  customColors.update(x => {
+    return {...x, [id]: { color }};
```

Comment:
This object looks exactly like an array. 
![image](https://user-images.githubusercontent.com/11016151/70849847-eb4e6d00-1e94-11ea-8106-3561221fd0fb.png)
The structure should be changed to an array. The function is called "add", which means it should append to the end. Currently, with the index being passed, you can only add 1 extra color for each variety

Reply:
Sure about the object??
![image](https://user-images.githubusercontent.com/29180358/70987334-b890b880-20d0-11ea-8f45-db822109bc27.png)
Anyway, it is passed to one variety just because it is logical. If we try to just extend the `colors` array, we'll get aloooooooot of color in there due to `input type=color` work and no ability to delete them from ui.

Reply:
I added a `console.log` right before that return with the same expression, and that's what I got.